tiatoolbox/models/engine/engine_abc.py

```diff
@@ -1,75 +1,165 @@
 from abc import ABC, abstractmethod
-from typing import List, Tuple, Union
 
-import numpy as np
+import torch.nn as nn
 
+from tiatoolbox.models.architecture import get_pretrained_model
 
-class IOConfigABC(ABC):
-    """Define an abstract class for holding predictor I/O information.
 
-    Enforcing such that following attributes must always be defined by
-    the subclass.
+class EngineABC(ABC):
+    """Abstract base class for engines used in tiatoolbox.
+
+    Args:
+        model (nn.Module):
+            Use externally defined PyTorch model for prediction with.
+            weights already loaded. Default is `None`. If provided,
+            `pretrained_model` argument is ignored.
+        pretrained_model (str):
```

Comment:
Remove `pretrained_model` argument and only use `model`.

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/635/commits/8faa23fa779975d14417da7f181ef2b17062f647